python/cudf/cudf/utils/cudautils.py

```diff
@@ -1,9 +1,11 @@
 # Copyright (c) 2018-2021, NVIDIA CORPORATION.
 from functools import lru_cache
 
+import cachetools
```

Comment:
Do we need a new dependency for this? Can we not use `functools.lru_cache`?

Reply:
I didn't see a way to control the key with `functools.lru_cache` - is there a way I have missed?

Reply:
Fair enough. Let's move forward as is for now but we should generally try to avoid adding new dependencies whenever possible unless they're needed.

Reply:
I added the `cachetools` dependency as I noticed that something that cuDF depends on seems to depend on it, so it was already installed in my environment... However, the relevant code is approximately 160 lines - would vendoring it be better?

Reply:
Nah this is fine. Wasn't aware this was already in the dependency tree.

Reply:
@gmarkall, @kkraus14, since this is already somewhere in our dependency tree, should it still be added as an explicit dependency in the [integration](https://github.com/rapidsai/integration) repo?

Reply:
@ajschmidt8 Would that be added to the `run` section in https://github.com/rapidsai/integration/blob/branch-0.18/conda/recipes/rapids-build-env/meta.yaml ? If that's intended to install all packages that cuDF might use, then I'd guess it should be (my answer is a bit tentative because I'm not particularly familiar with the integration repo).

Reply:
cool, i will open a PR to add it there

Reply:
PR added here: https://github.com/rapidsai/integration/pull/220